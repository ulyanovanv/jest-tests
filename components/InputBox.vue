<template>
  <div class="input-box">
    <span class="input-box_symbol">{{ currentCurrencySymbol() }}</span>
    <input class="input-box_input" id="valueNum" type="text" v-model="currentValue" />

    <select class="input-box_dropdown" v-model="currentCurrency">
      <option v-for="(el, index) in Object.keys(currencySymbol)" :key="index">{{ el }}</option>
    </select>
  </div>
</template>

<script>
import { mapGetters, mapActions } from "vuex";

export default {
  name: "input-box",

  data() {
    return {
      currencySymbol: {
        'EUR': '€',
        'USD': '$',
        'GBP': '£'
      }
    }
  },

  props: {
    inputbox: String
  },

  computed: {
    ...mapGetters({
      rates: "rates/getRates",
      currency_1: 'rates/getCurrency_1',
      currency_2: 'rates/getCurrency_2',
      value_1: 'rates/getValue_1',
      value_2: 'rates/getValue_2',
    }),

    currentCurrency: {
      get: function() {
        return this.inputbox == 1 ? this.currency_1 : this.currency_2
      },
      set: function(value) {
        this.changeOfCurrency({
          inputBox: this.inputbox,
          value: value
        })
      }
    },

    currentValue: {
      get: function() {
        return this.inputbox == 1 ? this.value_1 : this.value_2
      },
      set: function(value) {
        this.changeOfValue({
          inputBox: this.inputbox,
          value: value
        })
      }
    }
  },

  methods: {
    ...mapActions({
      changeOfCurrency: 'rates/changeOfCurrency',
      changeOfValue: 'rates/changeOfValue'
    }),

    currentCurrencySymbol() {
      return this.currencySymbol[this.currentCurrency];
    }
  }
}
</script>
