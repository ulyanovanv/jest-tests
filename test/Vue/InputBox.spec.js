import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import InputBox from '../../components/InputBox.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('InputBox', () => {
  let getters, store

  getters = {
    'rates/getRates': () => {},
    'rates/getCurrency_1': () => 'EUR',
    'rates/getCurrency_2': () => 'USD',

    'rates/getValue_1': () => 1,
    'rates/getValue_2': () => 1.11
  }

  store = new Vuex.Store({
    getters
  })

  test('is a Vue instance', () => {
    const wrapper = shallowMount(InputBox, { store, localVue })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('should render content correctly', () => {
    const wrapper = shallowMount(InputBox, { store, localVue })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  // how to access store.getter method
  test('getter rates/getCurrency_1 has to return "EUR" value', () => {
    expect(getters['rates/getCurrency_1']()).toBe('EUR')
  })

  test('test computed value currentCurrency', () => {
    const commit = jest.fn();
    const wrapper = shallowMount(InputBox, { store, localVue })
    wrapper.setData({ currency_1: "EUR", currency_2: "USD" });
    wrapper.setProps({ inputbox: 1 });
    expect(wrapper.vm.currentCurrency).toBe("EUR")
    //how to use computed set() hier?
  })
})
