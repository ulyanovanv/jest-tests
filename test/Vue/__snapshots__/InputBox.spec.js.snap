// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox should render content correctly 1`] = `
<div
  class="input-box"
>
  <span
    class="input-box_symbol"
  >
    $
  </span>
   
  <input
    class="input-box_input"
    id="valueNum"
    type="text"
  />
   
  <select
    class="input-box_dropdown"
  >
    <option>
      EUR
    </option>
    <option>
      USD
    </option>
    <option>
      GBP
    </option>
  </select>
</div>
`;
